src/librbd/ImageWatcher.cc

```diff
@@ -572,12 +572,12 @@ template <typename I>
 void ImageWatcher<I>::schedule_request_lock(bool use_timer, int timer_delay) {
   ceph_assert(ceph_mutex_is_locked(m_image_ctx.owner_lock));
 
-  if (m_image_ctx.exclusive_lock == nullptr) {
-    // exclusive lock dynamically disabled via image refresh
+  // exclusive lock dynamically disabled via image refresh or
+  // race with image task
```

Comment:
Nit: this comment doesn't make much sense to me -- "image task" is rather vague...  Since this just repeats the check in `notify_request_lock` to avoid unnecessary work:
```suggestion
  // see notify_request_lock()
```